### PR TITLE
Add more robust SW registration snippet

### PR DESCRIPTION
**Pull request by addyosmani**
Addresses part of the feedback from https://github.com/vuejs/pwa/issues/12 with respect to better detecting secure origins are being used. It effectively uses a similar setup to what we originally had for Web Starter Kit.

r @jeffposnick

### CODE REVIEW

File: template/index.html [hunk 1]
```diff
@@ -26,16 +26,59 @@
...
+            window.location.hostname === '[::1]' ||
+            // 127.0.0.1/8 is considered localhost for IPv4.
+            window.location.hostname.match(
+              /^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/
+            )
+          );
+
+        if ('serviceWorker' in navigator &&
+            (window.location.protocol === 'https:' || isLocalhost)) {
+          navigator.serviceWorker.register('service-worker.js')
```
Comment (jeffposnick) [inline template/index.html]: It's usually a good idea to wrap the `register()` in a `window.onload`, as per https://developers.google.com/web/fundamentals/instant-and-offline/service-worker/registration#conclusion

If there's a "better" event to listen for that's exposed as part of the Vue framework, like an event that's fired after all all the custom framework logic has completed, that would be another option to wrap it in.

**Comment by addyosmani** [comment]
@jeffposnick PTAL. Added the delay to post onload.

**Comment by jeffposnick** [comment]
👍

**Comment by addyosmani** [comment]
Thanks for reviewing!

**Pull request merged**